### PR TITLE
Fix horizontal scroll and last element selection in the Tree

### PR DESCRIPTION
**Pull request by istarkov**
## Description

ref #2151 

https://discord.com/channels/955905230107738152/1143789794426638377

Also made that scrollbars are not intersected
image

## Steps for reproduction

Add a lot of elements into the tree.
Be sure last element has a long long label.

See it can be selected, 
See both scrolls work

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-builder/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: apps/builder/app/dashboard/header.tsx [hunk 1]
```diff
@@ -67,9 +67,9 @@ export const Header = ({ user }: { user: User }) => {
       className={containerStyle()}
     >
       <WebstudioIcon width={30} height={23} />
-      <Flex gap="1" align="center">
+      <div>
```
Comment (kof) [inline apps/builder/app/dashboard/header.tsx:70]: huh, why this change in dashboard?

File: packages/design-system/src/components/scroll-area.tsx [hunk 2]
```diff
@@ -32,15 +50,31 @@ const ScrollAreaScrollbar = styled(Scrollbar, {
...
+      both: {
+        "&[data-orientation=vertical]": {
+          marginBottom: theme.spacing[4],
+        },
+        '&[data-orientation="horizontal"]': {
+          marginRight: theme.spacing[4],
+        },
+      },
+      horizontal: {},
+      vertical: {},
```
Comment (kof) [inline packages/design-system/src/components/scroll-area.tsx:76]: souldn't you have the same thing as in both but for horizontal and vertical?

Comment (kof) [reply packages/design-system/src/components/scroll-area.tsx:76]: ah no, all good, this is space for when both are visible

File: packages/design-system/src/components/tree/tree.tsx [hunk 3]
```diff
@@ -288,6 +289,10 @@ export const Tree = <Data extends { id: string }>({
         onBlur={keyboardNavigation.handleBlur}
         onKeyDown={keyboardNavigation.handleKeyDown}
         onClick={keyboardNavigation.handleClick}
+        css={{
```
Comment (kof) [inline packages/design-system/src/components/tree/tree.tsx:292]: shoudln't this somehow automatically be prevented when content is wrapped in scrollable? we don't do marginRight for the right scroll either ... you don't want to have to define margin every time you wan to use scroll

File: packages/design-system/src/components/scroll-area.tsx (hunk 2 above)
Comment (istarkov) [reply packages/design-system/src/components/scroll-area.tsx:76]: image
Only this case need to be solved

File: packages/design-system/src/components/tree/tree.tsx (hunk 3 above)
Comment (istarkov) [reply packages/design-system/src/components/tree/tree.tsx:292]: In most cases we don't need this margin. Here I added it only to improve a little bit last element selectability

Comment (istarkov) [reply packages/design-system/src/components/tree/tree.tsx:292]: If you have nothing to select no need in margin

**Review by kof — APPROVED** [review]
